tensorflow/compiler/mlir/tosa/transforms/passes.h

```diff
@@ -17,18 +17,19 @@ limitations under the License.
 #define TENSORFLOW_COMPILER_MLIR_TOSA_TRANSFORMS_PASSES_H
 
 #include <memory>
+#include <unordered_set>
 
 #include "mlir/Pass/Pass.h"  // from @llvm-project
 
 namespace mlir {
 namespace tosa {
 
 void populateLegalizeTFPatterns(MLIRContext* ctx, RewritePatternSet& patterns);
-void populateLegalizeTFLPatterns(MLIRContext* ctx, RewritePatternSet& patterns);
+void populateLegalizeTFLPatterns(MLIRContext* ctx, RewritePatternSet& patterns, std::unordered_set<std::string> legalization_disable = {});
```

Comment:
Can we use a llvm::StringSet instead here?

Reply:
My rationale is that the data structure can be coming from an external library part of the code that does not depend on LLVM, so keeping it as a standard library data structure seems more flexible.

Reply:
I will very likely be called inside an MLIR pass. Maybe this pass will be initialized from external sources, but we can perform the conversion in the initialization if needed.

I also make a difference between `createXXXXPass` (happens "rarely") and `populateXXXXXPatterns` (often happen during pass execution, even though it can happen during pass initialization, but folks rarely do that)

Actually the best way to decouple from the storage is still to take a function_ref callback here:

`void populateLegalizeTFLPatterns(MLIRContext* ctx, RewritePatternSet& patterns, llvm::function_ref<bool(StringRef)> is_op_enabled = nullptr);`

Inside the `populateLegalizeTFLPatterns` function, you can just call it back: `if(is_op_enabled(OpName)) addPattern...`